lib/src/body.dart

```diff
@@ -52,9 +52,16 @@ class Body {
         contentLength = encoded.length;
         stream = Stream.fromIterable([encoded]);
       }
+    } else if (body is List<int>) {
+      // Avoid performance overhead from an unnecessary cast.
+      contentLength = body.length;
+      stream = Stream.value(body);
     } else if (body is List) {
       contentLength = body.length;
-      stream = Stream.fromIterable([body.cast()]);
+      stream = Stream.value(body.cast());
```

Comment:
If this is likely to be iterated over multiple times, you may be better to eagerly copy the list with `List.from` instead of using `cast`.

Reply:
My hunch is this isn't a problem. I don't expect multiple iterations in most cases. Usually when we would need this cast it would be a `Response` which will be passed to `dart:io` which I expect iterates once. Otherwise, when this stream is getting passed to user code (where it may be iterated more than once) I'd expect the optimizations in this PR should kick in since it will most often be a `Stream<List<int>>` coming from `dart:io` to start with.